packages/services/src/market-data/market-data.service.ts

```diff
@@ -0,0 +1,148 @@
+import {
+  Brc20CryptoAssetInfo,
+  BtcCryptoAssetInfo,
+  FungibleCryptoAssetInfo,
+  MarketData,
+  NativeCryptoAssetInfo,
+  RuneCryptoAssetInfo,
+  Sip10CryptoAssetInfo,
+  StxCryptoAssetInfo,
+  createMarketData,
+  createMarketPair,
+} from '@leather.io/models';
+import {
+  baseCurrencyAmountInQuote,
+  calculateMeanAverage,
+  convertAmountToFractionalUnit,
+  createMoney,
+  getPrincipalFromContractId,
+  initBigNumber,
+  isDefined,
+  isFulfilled,
+} from '@leather.io/utils';
+
+import { AlexSdkClient } from '../infrastructure/api/alex-sdk/alex-sdk.client';
+import { BestInSlotApiClient } from '../infrastructure/api/best-in-slot/best-in-slot-api.client';
+import { BinanceApiClient } from '../infrastructure/api/binance/binance-api.client';
+import { CoincapApiClient } from '../infrastructure/api/coincap/coincap-api.client';
+import { CoinGeckoApiClient } from '../infrastructure/api/coingecko/coingecko-api.client';
+
+export interface MarketDataService {
+  getMarketData(asset: FungibleCryptoAssetInfo, signal?: AbortSignal): Promise<MarketData>;
+  getBtcMarketData(signal?: AbortSignal): Promise<MarketData>;
+  getStxMarketData(signal?: AbortSignal): Promise<MarketData>;
+  getSip10MarketData(asset: Sip10CryptoAssetInfo): Promise<MarketData>;
+  getBrc20MarketData(asset: Brc20CryptoAssetInfo, signal?: AbortSignal): Promise<MarketData>;
+  getRuneMarketData(asset: RuneCryptoAssetInfo, signal?: AbortSignal): Promise<MarketData>;
+}
+
+export function createMarketDataService(
+  bestInSlotApiClient: BestInSlotApiClient,
+  coinGeckoApiClient: CoinGeckoApiClient,
+  coincapApiClient: CoincapApiClient,
+  binanceApiClient: BinanceApiClient,
+  alexSdkClient: AlexSdkClient
+): MarketDataService {
+  async function getMarketData(asset: FungibleCryptoAssetInfo, signal?: AbortSignal) {
+    switch (asset.assetType) {
+      case 'nativeBtc':
+        return await getBtcMarketData(signal);
+      case 'nativeStx':
+        return await getStxMarketData(signal);
+      case 'sip10':
+        return getSip10MarketData(asset);
+      case 'rune':
+        return getRuneMarketData(asset, signal);
+      case 'brc20':
+        return getBrc20MarketData(asset, signal);
+      default:
+        throw Error('Market data not supported for asset type: ' + asset.assetType);
+    }
+  }
+
+  async function getBtcMarketData(signal?: AbortSignal) {
+    return await _getNativeAssetMarketData({ symbol: 'BTC' } as BtcCryptoAssetInfo, signal);
+  }
+
+  async function getStxMarketData(signal?: AbortSignal) {
+    return await _getNativeAssetMarketData({ symbol: 'STX' } as StxCryptoAssetInfo, signal);
+  }
+
+  async function _getNativeAssetMarketData(asset: NativeCryptoAssetInfo, signal?: AbortSignal) {
```

Comment:
What's the meaning of the _ naming?

Reply:
The `_` prefix is a convention that indicates a function (or variable) is private within the service - i.e. it's an internal helper and not part of the public interface.

Makes it clear that its not meant to be exported and is just an implementation 
detail (and should only called by other functions in the service).